### DISCUSSION TITLE
With Dialogic 2.0, is there anyway to customize the boxes?

### DISCUSSION THREAD

**Discussion by battlestoriesfan** [dialogic-godot/dialogic#1930] [Q&A]
I'm making this visual novel, and I have some ideas for customizeable boxes, with different patterns and perhaps looping animations to match each character. But I'm not finding any ways to customize the dialog boxes. Is there any way to do it?

And while we're at it, is there a way to make the choices and normal dialog boxes look different from each other?

**Comment by CakeVR** [comment]
When you head to the Character tab in Dialogic, you can set custom styles per character in the character settings.

Overall, you can design the entire look as you want if you make it a custom layout. 
Take a look here: https://dialogic-docs.coppolaemilio.com/documentation/styles-and-layouts/

Nonetheless, there is a rework happening about how Dialogic scenes are structured: https://github.com/coppolaemilio/dialogic/pull/1925

**Reply by battlestoriesfan** [reply]
I saw the style for each character. But What i mean is that, looking at it, doesn't seem that we can add a texture to the boxes or have custom animations or the like.

OH! I haven't seen that rework happening...But i already have quite a bit of my timeline made...hopefully the plug-in won't change too much...but it is interesting to hear about those changes!

**Reply by CakeVR** [reply]
You can do all of this. 
If you make a custom layout, you can change everything as you want.
The settings you see inside the style are nothing but exported variables visible for designers. You, as the custom layout creator, can extend these settings.

**Reply by battlestoriesfan** [reply]
OH! 
That sounds exactly like what I want to! None of the tutorials I found for Dialogic talked about that, so I didn't even know about that feature just yet.

Thanks! I'll be sure to take a look at it later

**Reply by CovetHiss** [reply]
In my case, I did duplicate the DialogicDefaultLayout scene. Then I realized most of the boxes are using panel with flat colors. If you want to change the design of these UI boxes, 1st go to the box you wish to change the design, and go to Inspector > Control section > Theme > Styles. The style will be flat color by default. If you already has a texture for these boxes, then change the styleBoxFlat to styleBoxTexture then drag in your texture.

this is how mine looks like. and it works pretty fine. If you are able to do this you dont have to rely on the exported settings in theme section because the settings there is kinda limited.
![DialogicDefaultLayout tscn - BoW - Godot Engine 11_30_2023 3_31_58 PM](https://github.com/coppolaemilio/dialogic/assets/110328712/997ba5b3-eeab-40d9-a5da-ed4b363f68d0)

**Reply by battlestoriesfan** [reply]
I tried to do what you did and took a look at it, and now I see what you mean, there really is EVERYTHING there. Honestly, I'd prefer, in the future, to just create a new style to export the variables, since then I'll be able to control everything. 

But i'm still a giant beginner with Godot as a whole, so that'll have to wait.

Still thanks for the answer too, I didn't know the default style was an actual scene! Although I should have expected

**Reply by Falkonsflight** [reply]
I dont kow why but I trying it for nearly 4 hours now and if I add the texture it only shows a black box instead. Tried everything and it just dont work properly.
